src/Form/FilePersisterServiceSettingsForm.php

```diff
@@ -145,10 +178,155 @@ public function buildForm(array $form, FormStateInterface $form_state) {
+          break;
+      }
+    }
+    if (!$valid) {
+      $warning_message = self::filePathErrorMessage('file', $scheme, $path);
+      $response->addCommand(new InvokeCommand('#edit-file-path', 'addClass',
+        ['error']));
+      $response->addCommand(new InvokeCommand('#edit-file-path', 'removeClass',
+        ['ok']));
+      $response->addCommand(new MessageCommand($warning_message, NULL,
+        ['type' => 'error', 'announce' => 'file path is not valid.']));
+
+    }
+    else {
+      $response->addCommand(new InvokeCommand('#edit-file-path', 'removeClass',
+        ['error']));
+      $response->addCommand(new InvokeCommand('#edit-file-path', 'addClass',
+        ['ok']));
+      $response->addCommand(new MessageCommand('Relative file path is valid for ' . $scheme . '://',
+        NULL, ['type' => 'status', 'announce' => 'file path is valid!']));
+
+    }
+    return $response;
+  }
+
+  /**
+   * Validate Object File Path
+   *
+   * @param  array  $form
+   * @param  \Drupal\Core\Form\FormStateInterface  $form_state
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   */
+  public function validateObjectFilePath(
+    array $form,
+    FormStateInterface $form_state
+  ) {
+    $response = new AjaxResponse();
+    $path = $form_state->getValue('object_file_path');
+    $scheme = $form_state->getValue('object_file_scheme');
+    if (empty($path)) {
+      $valid = TRUE;
+    }
+    else {
+      switch ($scheme) {
+        case 's3':
+          $valid = \Drupal::service('strawberryfield.utility')
+            ->s3FilePathIsValid($path);
+          break;
+        default:
+          $valid = \Drupal::service('strawberryfield.utility')
+            ->internalFilePathIsValid($scheme, $path);
+          break;
+      }
+    }
+    if (!$valid) {
+      $warning_message = self::filePathErrorMessage('object file', $scheme,
+        $path);
+      $response->addCommand(new InvokeCommand('#edit-object-file-path',
+        'addClass', ['error']));
+      $response->addCommand(new InvokeCommand('#edit-object-file-path',
+        'removeClass', ['ok']));
+      $response->addCommand(new MessageCommand($warning_message, NULL,
+        ['type' => 'error', 'announce' => 'object file path is not valid.']));
+
+    }
+    else {
+      $response->addCommand(new InvokeCommand('#edit-object-file-path',
+        'removeClass', ['error']));
+      $response->addCommand(new InvokeCommand('#edit-object-file-path',
+        'addClass', ['ok']));
+      $response->addCommand(new MessageCommand('Relative object file path is valid for ' . $scheme . '://',
+        NULL,
+        ['type' => 'status', 'announce' => 'object file path is valid!']));
+
+    }
+    return $response;
+  }
+
+  /**
+   * Utility function builds warning message string for file path validation
+   * functions.
+   *
+   * @param $type
+   * @param $scheme
+   * @param $path
+   *
+   * @return mixed
+   */
+  private function filePathErrorMessage($type, $scheme, $path) {
```

Comment:
See latest commit. Fairly major refactoring of a lot of this stuff since we are changing to using the same validation rule for S3 and local file systems.